src/components/post/Post.js

```diff
@@ -52,32 +52,33 @@ const styles = theme => ({
 });
 
 class Post extends Component {
-  constructor() {
-    super();
-    this.state = {
-      // title: "",
-      // authors: "",
-      // time: "",
-      // tags: [],
-      // summary: "",
-      // dataset: null,
-      // notebook: null,
-      // isSubmit: false,
-      // views: 167
-    };
-  }
+  // constructor() {
+  //   super();
+  //   this.state = {
+  //     // title: "",
+  //     // authors: "",
+  //     // time: "",
+  //     // tags: [],
+  //     // summary: "",
+  //     // dataset: null,
+  //     // notebook: null,
+  //     // isSubmit: false,
+  //     // views: 167
+  //   };
+  // }
 
   render() {
+    console.log("POST VIEW RENDERING")
     const { classes } = this.props;
     const { upload } = this.props.upload;
+
     let dataLinkButton;
     let dataLink;
-
     if (upload != null) {
       dataLink = "/" + upload.metadata.dataset;
 
       // if there is a dataset, display a link to it
-      if (upload.metadata.dataset != "") {
+      if (upload.metadata.dataset !== "") {
```

Comment:
for your implementation, if there is no dataset you don't get anything returned, so you would have to enhance this if statement, something like this:

|| upload.metadata.dataset === null

Reply:
fixed